### PR TITLE
Reduce build time by disabling explore

### PR DESCRIPTION
**Pull request by anwar6953**
The build time was significantly increased by this PR: https://github.com/caskdata/cdap/pull/4859
Disable explore in cdap-app-fabric, cdap-data-fabric, cdap-data-fabric-tests to reduce build time.

http://builds.cask.co/browse/CDAP-RBT605-4

### CODE REVIEW

File: cdap-client-tests/src/test/java/co/cask/cdap/client/QueryClientTestRun.java [hunk 1]
```diff
@@ -30,10 +30,12 @@
 import co.cask.cdap.proto.NamespaceMeta;
 import co.cask.cdap.proto.QueryResult;
 import co.cask.cdap.security.authentication.client.AccessToken;
+import co.cask.cdap.test.TestConfiguration;
```
Comment (bdmogal) [inline cdap-client-tests/src/test/java/co/cask/cdap/client/QueryClientTestRun.java]: unused

File: cdap-client-tests/src/test/java/co/cask/cdap/client/QueryClientTestRun.java [hunk 2]
```diff
@@ -30,10 +30,12 @@
...
 import co.cask.cdap.test.XSlowTests;
 import com.google.common.collect.Lists;
 import org.junit.Assert;
 import org.junit.Before;
+import org.junit.ClassRule;
```
Comment (bdmogal) [inline cdap-client-tests/src/test/java/co/cask/cdap/client/QueryClientTestRun.java]: ditto

File: cdap-standalone/src/test/java/co/cask/cdap/StandaloneTester.java [hunk 3]
```diff
@@ -61,6 +62,16 @@ protected void before() throws Throwable {
     cConf.setBoolean(Constants.Dangerous.UNRECOVERABLE_RESET, true);
     cConf.setBoolean(StandaloneMain.DISABLE_UI, true);
 
+    // Setup test case specific configurations.
+    // The system properties are usually setup by TestConfiguration class using @ClassRule
+    for (String key : System.getProperties().stringPropertyNames()) {
+      if (key.startsWith(TestConfiguration.PROPERTY_PREFIX)) {
+        String value = System.getProperty(key);
+        cConf.set(key.substring(TestConfiguration.PROPERTY_PREFIX.length()), System.getProperty(key));
+        LOG.info("Custom configuration set: {} = {}", key, value);
```
Comment (bdmogal) [inline cdap-standalone/src/test/java/co/cask/cdap/StandaloneTester.java]: debug?

File: cdap-client-tests/src/test/java/co/cask/cdap/client/common/ClientTestBase.java [hunk 4]
```diff
@@ -57,6 +58,9 @@
   public static final SingletonExternalResource STANDALONE = new SingletonExternalResource(new StandaloneTester());
 
   @ClassRule
+  public static final TestConfiguration CONFIG = new TestConfiguration("explore.enabled", "true");
```
Comment (bdmogal) [inline cdap-client-tests/src/test/java/co/cask/cdap/client/common/ClientTestBase.java]: is explore needed for all client tests?

File: cdap-standalone/src/test/java/co/cask/cdap/StandaloneTester.java [hunk 5]
```diff
@@ -61,6 +62,16 @@ protected void before() throws Throwable {
     cConf.setBoolean(Constants.Dangerous.UNRECOVERABLE_RESET, true);
     cConf.setBoolean(StandaloneMain.DISABLE_UI, true);
 
+    // Setup test case specific configurations.
+    // The system properties are usually setup by TestConfiguration class using @ClassRule
+    for (String key : System.getProperties().stringPropertyNames()) {
```
Comment (bdmogal) [inline cdap-standalone/src/test/java/co/cask/cdap/StandaloneTester.java]: this is unrelated, right? just extra logging?

File: cdap-standalone/src/test/java/co/cask/cdap/StandaloneTester.java (hunk 3 above)
Comment (anwar6953) [inline cdap-standalone/src/test/java/co/cask/cdap/StandaloneTester.java]: I think this will be seldomly used (i.e. for select confiugrations) that info is fine.
It's also info in TestBase, where I got this logic from (to attempt to also put the same functionality in StandaloneTester).
https://github.com/caskdata/cdap/blob/2b0fa17a72a508fa6aac0341fff8224609025b12/cdap-unit-test/src/main/java/co/cask/cdap/test/TestBase.java#L302-L308

File: cdap-client-tests/src/test/java/co/cask/cdap/client/common/ClientTestBase.java (hunk 4 above)
Comment (anwar6953) [inline cdap-client-tests/src/test/java/co/cask/cdap/client/common/ClientTestBase.java]: No, but since StandaloneTester is also a `TestRule`, I had to put it like this:
```java
@ClassRule
  public static final SingletonExternalResource STANDALONE = new SingletonExternalResource(new StandaloneTester());

  @ClassRule
  public static final TestConfiguration CONFIG = new TestConfiguration("explore.enabled", "true");
```

File: cdap-standalone/src/test/java/co/cask/cdap/StandaloneTester.java (hunk 5 above)
Comment (anwar6953) [inline cdap-standalone/src/test/java/co/cask/cdap/StandaloneTester.java]: this is related. The test cases specify configuration parameters and values, and this is where/how they get detected and put in the CConfiguration

Comment (bdmogal) [inline cdap-standalone/src/test/java/co/cask/cdap/StandaloneTester.java]: oh! I missed the conf.set !! ;) :+1:

**Comment by bdmogal** [comment]
LGTM pending build.

**Comment by anwar6953** [comment]
Build is 164 minutes on release/3.3 currently, and 130 minutes in this branch.
Build passed; merging.
